### PR TITLE
Run integration tests nightly

### PR DESCRIPTION
**Pull request by michaelkaye**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [X] Technical
- [ ] Other :

## Content

Moved sanity test and integration test into a single workflow, triggering that workflow nightly at 8pm, and also on `synchronize` of any branch called `release/*`.

## Motivation and context
Integration tests are taking > 2h to run, and we are not taking them on board when merging PRs.

Moving to a nightly run reduces wasted work (freeing up CI resources to do other tasks, faster) and will stand as a place to run the tests until we have them running reliably and quickly, and can re-introduce them to the PR flow.

## Tests

Hard to test: on purpose this PR does NOT run the tests :)

After merging to develop, trigger a nightly test run manually and check ~3h later if the notification arrives.

## Tested devices

GHA
## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [X] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Review by ouchadam — APPROVED** [review]
merging together for the nightly makes sense to me and we'll still be able to see the individual steps if needed

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -1,21 +1,25 @@
-name: Integration Tests
+name: Nightly Tests
 
 on:
-  pull_request: { }
   push:
-    branches: [ main, develop ]
+    branches: [ release/* ]
```
Comment (bmarty) [inline .github/workflows/nightly.yml:5]: does it mean that it will not run on develop branch, but only on release branches?

**Review by bmarty — COMMENTED** [review]
Just one question, else LGTM

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (ouchadam) [reply .github/workflows/nightly.yml:5]: I _believe_ crons are set to the latest commit on the default branch, this would just be for pushes

Comment (michaelkaye) [reply .github/workflows/nightly.yml:5]: All pushes to release branches; cron on "the default branch"

Comment (michaelkaye) [reply .github/workflows/nightly.yml:5]: So the only extra we could have could be on push to the develop branch?

**Comment by michaelkaye** [comment]
Going to merge for the 20:00 run; we can revert if that causes trouble.

**Comment by michaelkaye** [comment]
Nightly

This is the option in github actions to cause a nightly run to occur out of the standard schedule, if we want to force a test to start after a large change has been merged, or whatever.

**Pull request merged**